tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/Summary.java

```diff
@@ -347,6 +356,25 @@ private void makeAutoPriorityItemsSnapshot() throws IOException, JSONException {
         log("Automatic Summary Snapshot, finished; status = " + sr.status);
     }
 
+    private boolean autoSnapTooLong(long startMillis) {
+        final long elapsedMillis = System.currentTimeMillis() - startMillis;
+        final long elapsedMinutes = TimeUnit.MINUTES.convert(elapsedMillis, TimeUnit.MILLISECONDS);
+        if (elapsedMinutes > AUTO_SNAP_MAX_DURATION_MINUTES) {
+            log("Automatic Summary Snapshot timed out, stopping");
```

Comment:
should have some way to log this… maybe SurveyLog.logException since those are tracked?

Reply:
it will show up in console.log -- what kind of "tracked" do you mean?

Reply:
For later, the new `SurveyMetrics` class tracks number of exceptions thrown.  We could track metrics around how long snapshots take to run, and a timestamp of when the last snapshot was taken, with alerts if something is out of range

Reply:
it will show up in console.log -- what kind of "tracked" do you mean?